components/identity.oauth.uma.permission.service/src/main/java/org/wso2/carbon/identity/oauth/uma/service/internal/PermissionServiceComponent.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.uma.service.internal;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.wso2.carbon.identity.oauth.uma.service.PermissionService;
+import org.wso2.carbon.identity.oauth.uma.service.impl.PermissionServiceImpl;
+
+/**
+ * Service component for UMA permission endpoint.
+ */
+@Component(name = "org.wso2.carbon.identity.oauth.uma.service.internal.PermissionServiceComponent",
+           immediate = true)
+public class PermissionServiceComponent {
+
+    @Activate
+    protected void activate(BundleContext bundleContext) {
+
+        bundleContext.registerService(PermissionService.class.getName(), new PermissionServiceImpl(), null);
```

Comment:
Do this within a try catch (throwable) block. If something goes wrong it will avoid going through an activate loop